source/config/config.go

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate paramgen -output=paramgen.go Config
+
+package config
+
+import (
+	"strings"
+
+	"github.com/conduitio-labs/conduit-connector-redshift/common"
+)
+
+// Config contains source-specific configurable values.
+type Config struct {
+	common.Configuration
+	// OrderingColumn is a name of a column that the connector will use for ordering rows.
+	OrderingColumn string `json:"orderingColumn" validate:"required"`
+	// Snapshot is the configuration that determines whether the connector
+	// will take a snapshot of the entire table before starting cdc mode.
+	Snapshot bool `json:"snapshot" default:"true"`
+	// BatchSize is a size of rows batch.
+	BatchSize int `json:"batchSize" default:"1000"`
+}
+
+// Validate executes manual validations beyond what is defined in struct tags.
+func (c *Config) Validate() error {
+	// c.DSN has required validation handled in struct tag.
+
+	// c.Table required validation is handled in stuct tag
+	// handling "lowercase", "excludesall= " and "lte=127" validations.
+	if c.Table != strings.ToLower(c.Table) {
+		return common.NewLowercaseError(ConfigTable)
+	}
+	if strings.Contains(c.Table, " ") {
+		return common.NewExcludesSpacesError(ConfigTable)
+	}
+	if len(c.Table) > common.MaxConfigStringLength {
+		return common.NewLessThanError(ConfigTable, common.MaxConfigStringLength)
+	}
+
+	// c.KeyColumns handling "lowercase", "excludesall= " and "lte=127" validations.
+	for _, v := range c.KeyColumns {
+		if v != strings.ToLower(v) {
+			return common.NewLowercaseError(ConfigKeyColumns)
+		}
+		if strings.Contains(v, " ") {
+			return common.NewExcludesSpacesError(ConfigKeyColumns)
+		}
+		if len(v) > common.MaxConfigStringLength {
+			return common.NewLessThanError(ConfigKeyColumns, common.MaxConfigStringLength)
+		}
+	}
+
+	// c.OrderingColumn handling "lowercase", "excludesall= " and "lte=127" validations.
+	if c.OrderingColumn != strings.ToLower(c.OrderingColumn) {
+		return common.NewLowercaseError(ConfigOrderingColumn)
+	}
+	if strings.Contains(c.OrderingColumn, " ") {
+		return common.NewExcludesSpacesError(ConfigOrderingColumn)
+	}
+	if len(c.OrderingColumn) > common.MaxConfigStringLength {
+		return common.NewLessThanError(ConfigOrderingColumn, common.MaxConfigStringLength)
+	}
+
+	// c.BatchSize handling "gte=1" and "lte=100000" validations.
```

Comment:
these validations can be added to line 34 using paramgen to validate, https://github.com/ConduitIO/conduit-commons/tree/main/paramgen#:~:text=lt%20or%20less%2Dthan%3A%20takes,greater%20than%20the%20value%20provided.

Reply:
Updated. 
It is now `validate:"gt=0,lt=100001"`. 
But, I think it should be `validate:"gte=1,lte=100000"` (inclusive of the min and max values).

Reply:
we don't have those options for paramgen, it's ok to just change the limits with `gt` and `lt`